include/zephyr/shell/shell_mqtt.h

```diff
@@ -12,7 +12,7 @@
 #include <zephyr/net/socket.h>
```

Comment:
It's more of a subjective decision, none of us felt extremely strongly about it, but the placement feels more natural as opposed to it being in net/lib, since it is not really a library but a subsystem. I didn't include that reasoning in the commit since it is more subjective, especially with how Zephyr is organized in general. There isn't, to my knowledge, an objective argument for one location or another